pkgs/os-specific/linux/mdadm/default.nix

```
@@ -38,6 +38,17 @@ stdenv.mkDerivation rec {
         *.rules
   '';
 
+  postInstall = ''
+    install -D -m 755 misc/mdcheck ''${out}/bin/mdcheck
+    wrapProgram $out/bin/mdcheck \
+      --prefix PATH : ''${out}/bin:${lib.makeBinPath [ coreutils util-linux findutils gnugrep ]}
+
+    sed -i \
+      -e "s@/usr/share/mdadm/mdcheck@''${out}/bin/mdcheck@" \
+      -e "s@ExecStartPre=-/usr/lib/mdadm/mdadm_env.sh@@" \
+      ''${out}/lib/systemd/system/mdcheck* ''${out}/lib/systemd/system/mdmonitor*
```

```suggestion
      --prefix PATH : $out/bin:${lib.makeBinPath [ coreutils util-linux findutils gnugrep ]}

    sed -i \
      -e "s@/usr/share/mdadm/mdcheck@$out/bin/mdcheck@" \
      -e "s@ExecStartPre=-/usr/lib/mdadm/mdadm_env.sh@@" \
      $out/lib/systemd/system/mdcheck* $out/lib/systemd/system/mdmonitor*
```

If the module adds the mdadm package globally we don't need to prefix $out/bin